language/merge.js

```diff
@@ -1,18 +1,24 @@
 module.exports = function ( childRules, parentRules ) {
-	// Start off with a standard merge
-	const mergedRules = Object.assign( {}, parentRules, childRules );
+	// Only rules are preserved
+	const mergedRules = { rules: {} };
+
+	// Merge rules
+	Object.assign( mergedRules.rules, parentRules.rules, childRules.rules );
 
 	// For the specified keys, concatenate the lists
 	[ 'no-restricted-syntax', 'no-restricted-properties' ].forEach( function ( key ) {
-		// If either is unset, return the other
-		if ( !parentRules.rules[ key ] ) {
+		if ( !mergedRules.rules[ key ] ) {
+			// If both are unset, do nothing
+			return;
+		} else if ( !parentRules.rules[ key ] ) {
```

Comment:
These to no-parent and no-child branches seem redundant, if I understand correctly, the below assignment results in the same outcome as the main merge above already did (a no-op). Is that right? Maybe I'm missing something...

If so, then these two branches could be removed, with the last `else` becoming an `else if` for `parent && child`. Alternatively, it looks like maybe that branch would work fine unconditionally as well, given it already has fallbacks for both of them being absent. Those fallbacks are currently inaccessible given these two earlier branches.

Reply:
Yeah - it wasn't redundant when I didn't hardcode "error"